src/eosjs2-api.ts

```diff
@@ -120,14 +119,7 @@ export class Api {
         };
     }
 
-    async pushTransaction({ signatures, serializedTransaction }: PushTransactionArgs): Promise<any> {
-        return await this.rpc.push_transaction({
-            signatures,
-            serializedTransaction,
-        });
-    }
-
     async signAndPushTransaction(transaction: any) {
-        return await this.pushTransaction(await this.signTransaction(transaction));
+        return await this.rpc.push_transaction(await this.signTransaction(await this.fillTaposFields(transaction)));
```

Comment:
Wouldn't we want the API to still wrap the RPC call to push_transaction? With this change, there is no function to manually push signatures and a serialized transaction using the eosjs2 api.
Also, I was under the impression that the fillTapos should always be called with signTransaction so the format of the transaction will be correct every time, but if that is not the case then removing it from being called from within signTransaction makes sense.

Reply:
After our discussion, it seems like we still have some decisions to make with overall design before merging this pull request
I just checked the calendar and it looks as if Brandon will will be returning tomorrow.  I will be out for the first half of the day moving, but we could continue this conversation and make a more informed decision tomorrow.